### PR TITLE
Excluded given taxonomy for mappings in checking missing mappings

### PR DESCRIPTION
**Pull request by Catherine-Fu**
### What are you trying to accomplish?
1. enables users to specify which product category IDs should be excluded when checking missing mappings 
2. exclude specified product category IDs when checking missing mappings 
3. ensure category IDs won't be presented in the rules input and `unmapped_product_category_ids` at the same time

### Testing -1 
Exclude product category IDs from `unmapped_product_category_ids` when checking missing mappings

#### Before adding product category ids `ap-2-42-12` and `aa-1-2-9-13` to `unmapped_product_category_ids`
before-1

#### After adding product category ids `ap-2-42-12` and `aa-1-2-9-13 to `unmapped_product_category_ids`
after-2

### Testing -2 
Ensure category IDs won't be presented in the rules input and `unmapped_product_category_ids` at the same time.

adding `vp-2-3-4 ` to `unmapped_product_category_ids` and run `bin/rails test test/integration/mapping_validation_test.rb`: 

test-2

### CODE REVIEW

File: data/integrations/google/2021-09-21/excluded_taxonomy_for_mappings.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- input_taxonomy: shopify/2024-07
```
[the remark this answers is not in the document]
Comment (Catherine-Fu) [reply data/integrations/google/2021-09-21/excluded_taxonomy_for_mappings.yml]: I deleted this file and `unmapped_product_category_ids` will be added to `from_shopify.yml` or `to_shopify.yml` to specify which category ids should be excluded when checking missing mappings

File: data/integrations/google/2021-09-21/mappings/from_shopify.yml [hunk 2]
```diff
@@ -51288,3 +51288,109 @@ rules:
     product_category_id:
     - '5644'
 
+unmapped_product_category_ids:
```
[the remark this answers is not in the document]
Comment (Catherine-Fu) [reply data/integrations/google/2021-09-21/mappings/from_shopify.yml:51291]: This helps bypass CI checks of missing mappings from unstable Shopify taxonomy and we can easily constrain it to only check a specific version of Shopify taxonomy(e.g. 2024-07) once we have relevant versioned Shopify taxonomy ready for reading in running.  I can backlog a ticket to keep track of it.

**Comment by Catherine-Fu** [comment]
> Can we add something to explain about unmapped_product_category_ids to README?

Added.

File: data/integrations/README.md [hunk 3]
```diff
@@ -117,3 +117,28 @@ For the example above, `dist/en/integrations/all_mappings.json` would contain th
 }
 ```
 </details>
+
+#### Unmapped Product Categories
```
[the remark this answers is not in the document]
Comment (Catherine-Fu) [reply data/integrations/README.md:121]: I added [this section ](https://github.com/Shopify/product-taxonomy/pull/250/files#diff-5845eff92d1c8ab0648808937d99542d10d54f06a1806ade66195f26508a76fdR145)to explain `how unmapped_product_category_ids affects the final distribution mapping files`

**Comment by Catherine-Fu** [comment]
> [Revert "Excluded given taxonomy for mappings in checking missing mappings"](https://github.com/Shopify/product-taxonomy/pull/256)

In a following PR, we'll adjust Shopify taxonomy version `shopify/2024-07` to `shopify/unstable` for all mappings in directory `/data/integrations`.  Directory `/data/integrations` will become working directory for new mappings and directory `dist/` will hold previous versioned mappings.  

The CI validator will check against the unstable or latest Shopify taxonomy for new mappings and skip previous frozen versioned mappings.  cc: @elsom25

**Comment by elsom25** [comment]
>Directory /data/integrations will become working directory for new mappings and directory dist/ will hold previous versioned mappings.

I'm okay with keeping old versions in data, but we'll likely need to add some annotation/targeting to make sure its clear what version they target? Or we can drop them if y'all feel comfortable with only have dist variants of these things

**Pull request merged**